crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -139,16 +141,24 @@ impl Diagnostic {
         }
 
         if self.max_history_length > 1 {
-            if self.history.len() == self.max_history_length {
+            if self.history.len() >= self.max_history_length {
                 if let Some(removed_diagnostic) = self.history.pop_front() {
-                    self.sum -= removed_diagnostic.value;
+                    if !removed_diagnostic.value.is_nan() {
+                        self.sum -= removed_diagnostic.value;
+                    }
                 }
             }
 
-            self.sum += measurement.value;
+            if measurement.value.is_nan() {
```

Comment:
This logic seems backards.

Reply:
We probably actually want .is_finite.

Reply:
Forgot to prefix the condition with a !. But it's been switched to is_finite.